syntax = "proto3";

package mlops_model_analysis;

message ModelSpec {
    string name = 2;
    string model_ver = 3;
    repeated string label_keys = 4;
    repeated string prediction_keys = 5;
    repeated string prediction_prob_keys = 6;
}

message MetricConfig {
    string metric_name = 1;
    string config = 2;
}

message MetricSpec {
    repeated MetricConfig metrics = 1;    
}

message ModelScore {
    string score_name = 1;
    string report_column = 2;
    string report_row = 3;
    float threshold = 4;
}

message EvalConfig {
    ModelSpec model_spec = 2;
    MetricSpec metric_spec = 3;
    ModelScore model_score = 4;
}